client/domain/CodeSubmissionObjectFactory.js

```diff
@@ -74,6 +74,21 @@ tie.factory('CodeSubmissionObjectFactory', [
       }
     };
 
+    CodeSubmission.prototype.advanceImports = function() {
+      var pattern = new RegExp('^\\ {4}import\\ \\w+$');
+      var insertPos = 0;
+      for (var i = 0; i < this._preprocessedCodeLines.length; i++) {
+        if (pattern.test(this._preprocessedCodeLines[i])) {
+          var importLine = this._preprocessedCodeLines[i].slice(4);
```

Comment:
Could you make the numbers here constants of some kind? It's not clear what this code is doing just from looking at it, and that will cause us readability issues down the line.